VoodooInput/VoodooInputSimulator/VoodooInputSimulatorDevice.cpp

```diff
@@ -91,14 +89,28 @@ void VoodooInputSimulatorDevice::constructReportGated(const VoodooInputEvent& mu
         if (transducer->type == VoodooInputTransducerType::STYLUS) {
             continue;
         }
-        
+
         // in case the obtained id is greater than 14, usually 0~4 for common devices.
         UInt16 finger_id = transducer->secondaryId % 15;
         if (!transducer->isTransducerActive) {
             touch_state[finger_id] = 0;
         } else {
             input_active = true;
         }
+    }
+    
+    for (int i = 0; i < multitouch_event.contact_count + 1; i++) {
```

Comment:
Why the second inner loop?

Reply:
@kprinssu: in the first loop we calculate input_active via all fingers.
In the second loop we use input_active disregarding for which finger input_active was detected.